unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendsController.cs

```diff
@@ -36,26 +36,82 @@ public class FriendsController : IFriendsController
-            i = new FriendsController(apiBridge);
-        }
+        private CancellationTokenSource controllerCancellationTokenSource;
+
+        private DataStore dataStore;
+
+        private bool useSocialApiBridge => dataStore.featureFlags.flags.Get().IsFeatureEnabled(USE_SOCIAL_CLIENT_FEATURE_FLAG);
 
-        public FriendsController(IFriendsApiBridge apiBridge)
+        public FriendsController(IFriendsApiBridge apiBridge, IRPCSocialApiBridge rpcSocialApiBridge, DataStore dataStore)
         {
             this.apiBridge = apiBridge;
-            apiBridge.OnInitialized += Initialize;
             apiBridge.OnFriendNotFound += FriendNotFound;
             apiBridge.OnFriendWithDirectMessagesAdded += AddFriendsWithDirectMessages;
             apiBridge.OnUserPresenceUpdated += UpdateUserPresence;
             apiBridge.OnFriendshipStatusUpdated += HandleUpdateFriendshipStatus;
             apiBridge.OnTotalFriendRequestCountUpdated += UpdateTotalFriendRequests;
             apiBridge.OnTotalFriendCountUpdated += UpdateTotalFriends;
-            apiBridge.OnFriendRequestsAdded += AddFriendRequests; // TODO (NEW FRIEND REQUESTS): remove when we don't need to keep the retro-compatibility with the old version
-            apiBridge.OnFriendRequestReceived += ReceiveFriendRequest;
+
+
+            this.socialApiBridge = rpcSocialApiBridge;
+
+            socialApiBridge.OnFriendAdded += AddFriend;
+            socialApiBridge.OnFriendRemoved += InternalRemoveFriend;
+            socialApiBridge.OnFriendRequestAdded += AddFriendRequest;
+            socialApiBridge.OnFriendRequestRemoved += RemoveFriendRequest;
+
+            this.dataStore = dataStore;
         }
 
-        private void Initialize(FriendshipInitializationMessage msg)
+        public void Initialize()
+        {
+            controllerCancellationTokenSource = new CancellationTokenSource();
+
+            if (useSocialApiBridge)
+            {
+                socialApiBridge.InitializeClient(controllerCancellationTokenSource.Token).Forget();
+                socialApiBridge.InitializeFriendshipsInformation(controllerCancellationTokenSource.Token).ContinueWith(this.InitializeFriendships).Forget();
+            }
+            else
+            {
+                apiBridge.OnInitialized += InitializeFriendships;
+                apiBridge.OnFriendRequestsAdded += AddFriendRequests; // TODO (NEW FRIEND REQUESTS): remove when we don't need to keep the retro-compatibility with the old version
+                apiBridge.OnFriendRequestReceived += ReceiveFriendRequest;
+            }
+        }
+
+        public void Dispose()
+        {
+            controllerCancellationTokenSource.Cancel();
+            controllerCancellationTokenSource.Dispose();
+        }
+
+        private void RemoveFriendRequest(string userId)
+        {
+            this.friendRequests.Remove(userId);
+        }
+
+        // TODO (Joni): Replace by AddFriendRequests, this is just for successful compilation
```

Comment:
Are these comments relevants? is there anything missing?

Reply:
They will be covered in future PRs